appcli/stack_manager/archive_manager.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python3
+# # -*- coding: utf-8 -*-
+
+"""
+Handles archiving and rotation of logs and files.
+________________________________________________________________________________
+
+Created by brightSPARK Labs
+www.brightsparklabs.com
+"""
+
+
+# -----------------------------------------------------------------------------
+# IMPORTS
+# -----------------------------------------------------------------------------
+
+# Standard imports.
+from typing import Literal, List, Union
+from pathlib import Path
+import datetime
+import tarfile
+import glob
+
+# Vendor imports.
+from pydantic import BaseModel
+
+# Local imports.
+from appcli.models.cli_context import CliContext
+from appcli.logger import logger
+
+
+# -----------------------------------------------------------------------------
+# CLASSES
+# -----------------------------------------------------------------------------
+
+
+class CompressRule(BaseModel):
+    """Rule to compress files into an archived file.
+    This will overwrite any archive file that already exists.
+    """
+
+    type: Literal["compress"] = "compress"
+    """Explicitly declare rule type."""
+
+    name: str
+    """Name of the rule."""
+
+    include_list: List[str] = ["[]"]
+    """Glob pattern of files/dirs in `data_dir` to include in the archive."""
```

Comment:
Does it just apply to the `data_dir`? I thought it can do `backups` and `config` as well?

Reply:
Just the `data` dir.
See below comment.